constraints.pro

```diff
@@ -0,0 +1,67 @@
+% "name" is required for all workspaces (including the root).
+\+ gen_enforced_field(WorkspaceCwd, 'name', null).
+
+% "description" is required for all workspaces.
+\+ gen_enforced_field(WorkspaceCwd, 'description', null).
+
+% "repository.type" must be "git" for all workspaces (including the root).
+gen_enforced_field(WorkspaceCwd, 'repository.type', 'git').
+
+% "repository.type" must start with "git@github.com:mcmire/" for all workspaces
+% (including the root).
+% TODO: Figure this out
+%atom_starts_with(Atom, Prefix) :-
+  %sub_atom(Atom, _, _, _, Prefix).
+%gen_enforced_field(WorkspaceCwd, 'repository.url', RepoUrl) :-
+  %atom_starts_with(RepoUrl, 'git@github.com:mcmire/').
+
+% "license" must be "MIT" for all publishable workspaces and unset for the
+% root.
+gen_enforced_field(WorkspaceCwd, 'license', 'MIT') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'license', null) :-
+  workspace_field(WorkspaceCwd, 'private', true).
+
+% "main" must be "dist/index.js" for publishable workspaces and unset for the
+% root.
+gen_enforced_field(WorkspaceCwd, 'main', './dist/index.js') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'main', null) :-
+  workspace_field(WorkspaceCwd, 'private', true).
+
+% "types" must be "dist/index.d.ts" for publishable workspaces and unset for the
+% root.
+gen_enforced_field(WorkspaceCwd, 'types', './dist/index.d.ts') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'types', null) :-
+  workspace_field(WorkspaceCwd, 'private', true).
+
+% "files" must be ["dist/"] for publishable workspaces and unset for the root.
+gen_enforced_field(WorkspaceCwd, 'files', ['dist/']) :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'files', null) :-
+  workspace_field(WorkspaceCwd, 'private', true).
+
+% "engines.node" must be ">=12.0.0" for all workspaces.
+gen_enforced_field(WorkspaceCwd, 'engines.node', '>=12.0.0').
+
+% "publishConfig.access" must be "public" for publishable workspaces and unset
+% for the root.
+gen_enforced_field(WorkspaceCwd, 'publishConfig.access', 'public') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'publishConfig.access', null) :-
+  workspace_field(WorkspaceCwd, 'private', true).
+
+% "publishConfig.registry" must be "https://registry.npmjs.org" for publishable
+% workspaces and unset for the root.
+gen_enforced_field(WorkspaceCwd, 'publishConfig.registry', 'https://registry.npmjs.org/') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'publishConfig.registry', null) :-
+  workspace_field(WorkspaceCwd, 'private', true).
+
+% The version of Jest used in the root must be synchronized across all packages
+% that list Jest.
+gen_enforced_dependency(WorkspaceCwd, 'jest', DependencyRange, DependencyType) :-
```

Comment:
This is neat! We should consider whether to synchronize more packages. Maybe all direct dependencies?

Reply:
Sure! Added in [5f6205c](https://github.com/MetaMask/controllers/pull/831/commits/5f6205c0cf8b82361e1169af5eb8f189fdaa65c8).